vendor/github.com/jacobsa/gcloud/gcs/read.go

```diff
@@ -138,6 +153,38 @@ func (b *bucket) NewReader(
 	return
 }
 
+
+// Custom function made to create a new reader using Storage Client Library.
+func NewReaderSCL(
+        ctx context.Context,
+        req *ReadObjectRequest, bucketName string) (rc io.ReadCloser, err error){
+	// Create a client if there is not one already created.
+        if sclClient == nil {
+		fmt.Println("Client Created")
+		sclClient, err = storage.NewClient(ctx)
+		if err != nil {
+			err = fmt.Errorf("Error in creating the client: %v", err)
+			return
+		}
+	}
+
+	// Initialising the starting offset and the length to be read by the reader.
+        start := int64((*req.Range).Start)
```

Comment:
Is it possible that the Range is not being set?

Reply:
I think the range will always be valid.
[https://github.com/GoogleCloudPlatform/gcsfuse/blob/eaf4a4d58ee31765a9fb239b2f672e29f70aa441/internal/gcsx/random_reader.go#L332](url)
Whenever the native NewReader method is being called (as above), the range is always passed, consisting of start and end parameters. Also the start and end variables are in turn formed from offset and length variables which are passed by the OS to the ReadFile method in fs.go. So I think the arguments passed by the OS should be correct, but anyways many checks are performed on offset, length, start and end variables before passing it to the NewReader method. So I think the range will always be valid.
Also I wrote edge cases of various ranges in my unit tests, and they were all passing.